src/cython/sktda/metrics.py

```diff
@@ -0,0 +1,226 @@
+"""
+@author: Mathieu Carriere
+All rights reserved
+"""
+
+import sys
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.metrics import pairwise_distances
+try:
+    from gudhi import bottleneck_distance
+    USE_GUDHI = True
+except ImportError:
+    USE_GUDHI = False
+    print("Gudhi not found: BottleneckDistance will return null matrix, and exact SlicedWassersteinDistance not available")
```

Comment:
You mention an exact version, but it doesn't seem to be available now. IIUC, we will add it later as a wrapper for the C++ function (that is perfectly fine). What will it look like? Is it an extra parameter for `__init__` or a different class? Just checking if this has an impact on the desired interface for the approximate class.

Reply:
I was thinking of including the possibility to call the class with `num_directions` = -1, which would mean: do exact computation. Although, you are right, it is not available right now. It is even worse than this: the last time I checked, the exact computation code in C++ was not working on an example I had, so I have to debug and fix the exact computation code

Reply:
Side note: if you find an issue with gudhi (including your own C++ code) that you cannot fix right away, it can be a good idea to file an issue explaining how to repoduce the error, so the information is not lost (even for yourself). Of course if you can fix it immediately, just a pull request is fine, no need for the issue.

Reply:
> `num_directions` = -1, which would mean: do exact computation

That sounds good to me (although I would also interpret 0 directions as exact, maybe anything <=0), but we should remove the documentation of this possibility as long as the functionality isn't there.